tools/benchmark.py

```diff
@@ -43,15 +45,40 @@ def import_data_from_gh_pages():
         write_json(data_file, [])  # writes empty json data
 
 
-def main(argv):
+def get_build_dir_from_argv(argv):
     if len(argv) == 2:
-        build_dir = sys.argv[1]
+        return sys.argv[1]
     elif len(argv) == 1:
-        build_dir = build_path()
+        return build_path()
     else:
         print "Usage: tools/benchmark.py [build_dir]"
         sys.exit(1)
 
+
+# run strace with test_args and record times a syscall record appears in out file
+# based on syscall_line_matcher. Should be reusable
+def count_strace_syscall(syscall_name, syscall_line_matcher, test_args):
+    f = tempfile.NamedTemporaryFile()
+    run(["strace", "-f", "-o", f.name, "-e", "trace=" + syscall_name] +
+        test_args)
+    return len(filter(syscall_line_matcher, f))
+
+
+def run_thread_count_tests(deno_path):
+    thread_count_args_map = {
+        "set_timeout": ["tests/004_set_timeout.ts", "--reload"]
+    }
+    thread_count_map = {}
+    for name, test_args in thread_count_args_map.items():
```

Comment:
I would remove the loop here. Just do the single test. Move this code to benchmark_test.py